lib/diagnostic.js

```diff
@@ -6,34 +6,54 @@
 // splits it into an array of normalized words
 // (uppercase strings without punctuation)
 // and returns that array.
-const normalizeWords = function (/* */) {
-  /* your response here */
+const getNormalizedWords = function getNormalizedWords (text) {
+  let words = text.split(/\s+/);
+  let normalizedWords = [];
+  for (let i = 0; i < words.length; i++) {
+    let word = words[i].replace(/\w+/, '');
+    normalizedWords[i] = word.toUpperCase();
+  }
+  return normalizedWords;
 };
 
 // 2. Write a function that takes a string argument
 // and returns an array of unique normalized words.
-const uniqueWords = function (/* */) {
-  /* your response here */
+const getUniqueWords = function getUniqueWords(text) {
+let uniqueWordsAsKeys = {};
+let normalizedWords = getNormalizedWords(text);
+for (let i = 0; normalizedWords.length; i++) {
```

Comment:
This will result in an infinite loop.  The running condition ```normalizedWords.length``` is interpreted as a truthy value and will continue to run the loop.